Library/Homebrew/cmd/commands.rb

```diff
@@ -8,51 +8,70 @@ module Homebrew
   module_function
 
   def commands
-    if ARGV.include? "--quiet"
-      cmds = internal_commands + external_commands
-      cmds += internal_developer_commands
-      cmds += HOMEBREW_INTERNAL_COMMAND_ALIASES.keys if ARGV.include? "--include-aliases"
-      puts Formatter.columns(cmds.sort)
-    else
-      # Find commands in Homebrew/cmd
-      puts "Built-in commands"
-      puts Formatter.columns(internal_commands)
-
-      # Find commands in Homebrew/dev-cmd
-      puts
-      puts "Built-in developer commands"
-      puts Formatter.columns(internal_developer_commands)
-
-      # Find commands in the path
-      unless (exts = external_commands).empty?
-        puts
-        puts "External commands"
-        puts Formatter.columns(exts)
+    # TODO: Goal is to not require this function at all.
+    command = CommandsCommand.new
+    command.check_for_errors
+    command.commands
+  end
+
+  class CommandsCommand < Command
+    def initialize
+      options do
+        cmd_name "commands"
+        desc "Show a list of built-in and external commands."
+        option "quiet", desc: "list only the names of commands without the header." do
+          option "include-aliases", desc: "the aliases of internal commands will be included."
```

Comment:
I think we should to *document* options as having a relationship, but that it might make sense to flatten this out in the DSL. 

Just like with mutually exclusive options, there's no depth/complexity limit to child/parent option relationships. Handling a depth of two *works*, but we don't want to get caught in a spiral of DSL spaghetti for 3-level dependencies (or mutual exclusivities).